### DISCUSSION TITLE
How to include section references in HTML, but not in PDF?

### DISCUSSION THREAD

**Discussion by roaldarbol** [quarto-dev/quarto-cli#11056] [Q&A] [answered]
### Description

Is it possible to include references for each section in HTML only? MyST Markdown does this by automatically placing section references in the footnotes which I think is a neat feature; however, in PDF I only want the references at the end of the book.

**Comment by mcanouil** [comment]
This has already been discussed several times.
A search will give you them quite easily.

- https://github.com/quarto-dev/quarto-cli/discussions/7642

Regarding the "conditional" part, see conditional contents and profiles in <https://quarto.org>.

**Reply by roaldarbol** [reply]
As I guess you realised (since you edited your answer), it had not in fact been asked multiple times before. The question was explicitly about the conditionality, which the above discussion doesn't mention at all. At least your pointer to conditional content was on point. But your default assumption that I didn't bother to look is simply condescending, and really not making this Discussion board out to be the place I'll want to ask questions again.

**Reply by mcanouil** [reply]
> As I guess you realised (since you edited your answer), it had not in fact been asked multiple times before. 

I can guarantee you it has but not that easy to find again as it might have been part of other discussions.

> Is it possible to include references for each section in HTML only?

This is exactly what the thread is about and what the `section-bibliographies` does.

> But your default assumption that I didn't bother to look is simply condescending, and really not making this Discussion board out to be the place I'll want to ask questions again.

I did not assume anything.
I apologise if my message came out wrong.

Note that I did not close on purpose and also why I had to edit my message initially (I don't close any Q&A discussions because default GitHub filter removes all closed discussions which makes it less easy to search for discussions).

**Reply by mcanouil** [reply, marked as answer]
Note that extensions/filters can be applied only on some formats

```yaml
format:
  html:
    filters:
      - section-bibliographies
  pdf: default
```

https://github.com/pandoc-ext/section-bibliographies

**Reply by roaldarbol** [reply]
> I can guarantee you it has but not that easy to find again as it might have been part of other discussions.

I guess that's exactly the point; if it's not easily discoverable, users can not be expected to find the information - and it might be better to summarise it in a Discussion that specifically targets *that* question.

> This is exactly what the thread is about and what the `section-bibliographies` does.

I disagree that it's what the thread is about, but combined with your extra piece of information (which again I didn't know of), that definitely should solve the problem! 😊

> I did not assume anything. I apologise if my message came out wrong.

Fair enough. Asking "Have you come across...?", or "Does this [...] answer your question?", instead of basically saying "Already discussed, easy to find" goes a long way for future reference.

**Reply by mcanouil** [reply]
> Fair enough. Asking "Have you come across...?", or "Does this [...] answer your question?", instead of basically saying "Already discussed, easy to find" goes a long way for future reference.

Thanks for the feedback. I do appreciate it and will keep trying to do better.

**Reply by roaldarbol** [reply]
No worries - and sorry for souring your morning, I know negative feedback isn't fun to get. Appreciate the forthcomingness. 👌

**Reply by roaldarbol** [reply]
> Note that I did not close on purpose and also why I had to edit my message initially (I don't close any Q&A discussions because default GitHub filter removes all closed discussions which makes it less easy to search for discussions).

Oh, just remembered you saying this - feel free to re-open; I closed it again!

**Reply by mcanouil** [reply]
> No worries - and sorry for souring your morning, I know negative feedback isn't fun to get. Appreciate the forthcomingness. 👌

I actually don't mind at all as long as I can improve myself with them ;)

> Oh, just remembered you saying this - feel free to re-open; I closed it again!

That's 100 % fine, it's only my way of using GitHub Discussions because of the default filter, in case users don't remove it to search through all discussions.

![image](https://github.com/user-attachments/assets/bfe8cd14-c069-4b55-8595-729856302132)

**Reply by roaldarbol** [reply]
Haha, yeah that was how I found out - I couldn't immediately find it when I went back looking for it. 😅 So please do re-open it. 😂

**Comment by cderv** [comment]
> Is it possible to include references for each section in HTML only? MyST Markdown does this by automatically placing section references in the footnotes which I think is a neat feature

What are you doing exactly today ? Are you using a footnote style CSL style ? Using something like 
````
format:
  html:
    csl: https://www.zotero.org/styles/chicago-fullnote-bibliography
````

would lead to reference in the footnote and benefiting hover 
![image](https://github.com/user-attachments/assets/7b2ec438-73c4-4588-980c-332fdc3e7e22)

You can add this CSL for HTML only and it won't apply to PDF. 

If you have example to make your question clearer, this would help be sure we understand correctly and so answer accordingly. Thanks.

**Reply by roaldarbol** [reply]
Thanks for the response and for re-opening the discussion. I'm in the process of migrating a large project to Quarto. Your response helped clarify that there are more options and so what information I *could* provide, which I didn't know. 

I'm merely using citations like normal (`@Name1900`) and coming from MyST, I'm used to [this type of rendering](https://mystmd.org/guide/citations#references) in HTML, but not in PDF. So I was looking for a way to achieve just that. I think that the conditional content is probably what I'm looking for, but thanks for the help. 

I tend to prefer high-level questions to kick off with to avoid "how do I make this super hacky thing I'm doing work", when there might have been a straight-forward way that was just poorly documented. If there are better channels for asking informal questions, I'd be happy to be informed of where - I thought the discussion section would be just that (especially under Q&A) since it's where the Quarto website points "Ask a Question".

**Reply by cderv** [reply]
> I'm used to [this type of rendering](https://mystmd.org/guide/citations#references) in HTML

Are you working on Quarto Book or on Quarto Website ? 

I believe for website you will have this feature part of Article Layout https://quarto.org/docs/authoring/appendices.html#overview

For books layout is different, and quarto chose to not process appendix this ways. So citation as footnotes does not come with the appendix section but through other means. 

> I tend to prefer high-level questions to kick off with to avoid "how do I make this super hacky thing I'm doing work", when there might have been a straight-forward way that was just poorly documented. If there are better channels for asking informal questions, I'd be happy to be informed of where - I thought the discussion section would be just that (especially under Q&A) since it's where the Quarto website points "Ask a Question".

Yes you did get it right. It is just with high level question I am always unsure if I understood correctly. It leads to generic answer to me and this is open to interpretation. I only ask to get more precise ideas about what you wanted to MyST, as we may also have been missing the feature; 

So keep them coming, Q&A section is ok. We're just here to help. Thanks for sharing your feedback.

**Reply by roaldarbol** [reply]
Good to know, thanks. I find that a bit of iteration doesn't hurt anyone and often leads to the best solution, so this ping-pong is definitely helpful. 

I'm working on a Quarto Book; that makes good sense - unfortunate decision I think, but hey, decisions have to be made and can't suit all scenarios. 😊
